src/models/location/location-data-mapper-impl.ts

```diff
@@ -42,11 +43,11 @@ export class LocationDataMapperImpl extends GenericDataMapper
     super.addRBAC([this.READ, this.READ_ALL], [AuthLevel.PARTICIPANT]);
   }
 
-  public delete(id: UidType): Promise<IDbResult<void>> {
+  public delete(object: Location): Promise<IDbResult<void>> {
```

Comment:
I think the issue here is that the ID for location is a number while UidType is a string. Perhaps a rethinking of that type definition is important instead of just passing in a location type. But I think it might work